onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -585,14 +599,19 @@ void CreateGenericIterableMLValue(PyObject* iterator, AllocatorPtr alloc, const
   }
 }
 
+// Setting `use_numpy_data_memory` to `true` will ensure that the underlying numpy array buffer is directly used
+// as the backing data buffer for the ORT Tensor where applicable (for numeric tensors)
+// The numpy object owns the memory and needs to be alive until the corresponding OrtValue is in scope
 void CreateGenericMLValue(const onnxruntime::InputDefList* input_def_list, const AllocatorPtr& alloc, const std::string& name_input,
-                          py::object& value, OrtValue* p_mlvalue) {
+                          py::object& value, OrtValue* p_mlvalue, bool parse_numpy_only,
+                          bool use_numpy_data_memory, MemCpyFunc mem_cpy_to_device) {
```

Comment:
Based on the implementation 'parse_numpy_only' seems to equate to the input having to be something that PyObjectCheck_Array returns true for. I'm assuming a python array would be return true as well though. Would 'accept_only_array' be clearer?

Reply:
There are no "native" python arrays AFAIK. There are only python lists which we are not supporting for OrtValue creation (for this version atleast). So the intent is to only parse numpy and not python lists. Would `accept_only_numpy_array` be okay ?

Reply:
I didn't realize we owned PyObjectCheck_Array. I'd rename that to PyObjectCheck_NumpyArray for clarity, and accept_only_numpy_array is good with that.

Reply:
Done